ocaml/xapi/xapi_pusb.ml

```diff
@@ -88,3 +88,59 @@ let scan_thread ~__context =
   in
   start_thread f;
   scan ~__context
+
+let get_sm_usb_path ~__context vdi =
+  try
+    let vdi_sc = Db.VDI.get_sm_config ~__context ~self:vdi in
+    List.assoc Xapi_globs.usb_path vdi_sc
+  with _ -> ""
+
+let set_passthrough_enabled ~__context ~self ~value =
+  Mutex.execute mutex (fun () ->
+    match value with
+     | true ->
+       (* Remove the vdi records which 'usb_path' in sm-config has the
+          same value with the field 'path' in PUSB. *)
+       let pusb_path = Db.PUSB.get_path ~__context ~self in
+       let udev_srs = Db.SR.get_refs_where ~__context ~expr:(Eq(Field "type", Literal "udev")) in
+       List.iter (fun sr ->
+         Db.VDI.get_refs_where ~__context ~expr:(Eq(Field "SR", Literal (Ref.string_of sr))) |>
+           List.iter (fun rf ->
+             try
+               if (get_sm_usb_path ~__context rf) = pusb_path then begin
+                 Xapi_vdi.forget ~__context ~vdi:rf
+               end;
+             with e ->
+               debug "Caught failure during remove vdi records.";
+               raise e
+             )
+       ) udev_srs;
+       debug "set passthrough_enabled %b" value;
```

Comment:
Same here - the value is either true or false but in in the log you no longer know for which USB devices this was done. If this is covered in other log messages already, the log message can be removed.